index.js

```diff
@@ -21,30 +22,20 @@ const exec = (cmd, args) => {
 	).filter(Boolean);
 };
 
-const install = (bin, useYarn) => () => execa(bin, ['install'])
-	.catch(err => {
-		if (err.code === 'ENOENT' && useYarn) {
-			return execa('npm', ['install']);
-		}
-		throw err;
-	});
-
 module.exports = (input, opts) => {
 	input = input || 'patch';
 
 	opts = Object.assign({
 		cleanup: true,
-		publish: true
-	}, opts, {
-		yarn: opts && opts.yarn !== false
-	});
+		publish: true,
+		yarn: hasYarn()
+	}, opts);
```

Comment:
This is not entirely correct. If they force yarn by using `np --yarn` while there is no lockfile, it will create a lockfile and I think we should prevent that. Instead I would do something like

```js
opts = Object.assign({
    cleanup: true,
    publish: true
}, opts, {
    yarn: opts && opts.yarn && hasYarn()
});
```